_internal/utils/mutate.ts

```diff
@@ -35,130 +54,136 @@ export const internalMutate = async <Data>(
   const revalidate = options.revalidate !== false
   const rollbackOnError = options.rollbackOnError !== false
 
-  // Serialize key
-  const [key] = serialize(_key)
-  if (!key) return
-
-  const [get, set] = createCacheHelper<
-    Data,
-    State<Data, any> & {
-      // The previously committed data.
-      _c?: Data
+  // If 2nd arg is key filter, return the mutation results of filtered keys
+  if (isFunction(_key)) {
+    const keyFilter = _key
+    const matchedKeys: Key[] = []
+    for (const state of cache.values()) {
```

Comment:
Since we store the `key` in cache state, now we need to use `values()` or `entries()` method to get the result. Not sure if it's a good design for `Cache` typing.

Or use `keys`, but it requires to call `cache.get(key)` here, maybe it's a heavy call.